src/category_theory/adjunction/adjoint_functor_theorems.lean

```diff
@@ -0,0 +1,304 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.adjunction.basic
+import category_theory.limits.shapes
+import category_theory.limits.preserves.basic
+import category_theory.limits.creates
+import category_theory.punit
+
+universes v u
+
+namespace category_theory
+
+variables {J : Type v}
+variables {C : Type u} [category.{v} C]
+
+open category limits
+
+@[derive decidable_eq, derive inhabited] inductive walking_parallel_chunk (J : Type v) : Type v
+| zero : walking_parallel_chunk
+| one : walking_parallel_chunk
+
+open walking_parallel_chunk
+
+/-- The type family of morphisms for the diagram indexing a (co)equalizer. -/
+@[derive decidable_eq] inductive walking_parallel_chunk_hom (J : Type v) :
+  walking_parallel_chunk J → walking_parallel_chunk J → Type v
+| id : Π X : walking_parallel_chunk.{v} J, walking_parallel_chunk_hom X X
+| line : Π (j : J), walking_parallel_chunk_hom zero one
+
+open walking_parallel_chunk_hom
+
+/-- Composition of morphisms in the indexing diagram for (co)equalizers. -/
+def walking_parallel_chunk_hom.comp :
+  Π (X Y Z : walking_parallel_chunk J)
+    (f : walking_parallel_chunk_hom J X Y) (g : walking_parallel_chunk_hom J Y Z),
+    walking_parallel_chunk_hom J X Z
+  | _ _ _ (id _)   h := h
+  | _ _ _ (line j) (id one) := line j.
+
+section
+local attribute [tidy] tactic.case_bash
+
+instance walking_parallel_pair_hom_category  :
+  small_category (walking_parallel_chunk J) :=
+{ hom  := walking_parallel_chunk_hom J,
+  id   := walking_parallel_chunk_hom.id,
+  comp := walking_parallel_chunk_hom.comp }
+
+@[simp]
+lemma walking_parallel_chunk_hom_id (X : walking_parallel_chunk J) :
+  walking_parallel_chunk_hom.id X = 𝟙 X :=
+rfl
+
+/-- `parallel_pair f g` is the diagram in `C` consisting of the two morphisms `f` and `g` with
+    common domain and codomain. -/
+def parallel_chunk {X Y : C} (f : J → (X ⟶ Y)) : walking_parallel_chunk J ⥤ C :=
+{ obj := λ x, walking_parallel_chunk.cases_on x X Y,
+  map := λ x y h, match x, y, h with
+  | _, _, (id _) := 𝟙 _
+  | _, _, (line j) := f j
+  end,
+  map_comp' :=
+  begin
+    rintro _ _ _ ⟨⟩ ⟨⟩;
+    { unfold_aux, simp; refl },
+  end }
+
+end
+lemma gaft_aux (C : Type u) [category.{v} C] [has_limits.{v} C] {ι : Type v} (B : ι → C)
```

Comment:
Please expand the name. I'm quite sure I wouldn't know what `gaft` means outside the context of this file.